src/assets/CollectiblesController.ts

```diff
@@ -235,18 +249,40 @@ export class CollectiblesController extends BaseController<
     contractAddress: string,
     tokenId: string,
   ): Promise<CollectibleMetadata> {
-    const tokenURI = this.getCollectibleApi(contractAddress, tokenId);
-    let collectibleInformation: ApiCollectible;
+    // Attempt to fetch the data with the proxy
+    let collectibleInformation: ApiCollectible | undefined =
+      await fetchWithErrorHandling(
+        // the third arg defaults to true to use the proxy
+        this.getCollectibleApi(contractAddress, tokenId),
+      );
 
-    /* istanbul ignore if */
-    if (this.openSeaApiKey) {
-      collectibleInformation = await handleFetch(tokenURI, {
-        headers: { 'X-API-KEY': this.openSeaApiKey },
-      });
-    } else {
-      collectibleInformation = await handleFetch(tokenURI);
+    // if an openSeaApiKey is set we should attempt to refetch calling directly to OpenSea
+    if (!collectibleInformation) {
+      if (this.openSeaApiKey) {
+        collectibleInformation = await fetchWithErrorHandling(
+          this.getCollectibleApi(contractAddress, tokenId, false),
```

Comment:
When adding boolean arguments to functions/methods, it can be difficult to understand at first glance what those booleans signify. What do you think about having this method take an object instead of positional arguments?

Reply:
Yeah I think that makes sense!

Reply:
done here: https://github.com/MetaMask/controllers/pull/805/commits/180b2802d9edef8ff9392b88ea3a3f76cc3373da